### PR TITLE
OrderWalkthrough refactor

### PR DESCRIPTION
**Pull request by jhawthorn**
This had some spurious failures, likely due to relying on the existing
state of the database. After this it no longer relies on the global
zone, existing shipping methods, or existing countries.

Hoping this solves whatever happened here https://circleci.com/gh/solidusio/solidus/787

### CODE REVIEW

**Comment by gmacdougall** [comment]
Much nicer. Great work.

:+1: :balloon:

File: core/lib/spree/testing_support/order_walkthrough.rb [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
 
     # A shipping method must exist for rates to be displayed on checkout page
-    unless Spree::ShippingMethod.exists?
-      FactoryGirl.create(:shipping_method).tap do |sm|
-        sm.calculator.preferred_amount = 10
-        sm.calculator.preferred_currency = Spree::Config[:currency]
-        sm.calculator.save
-      end
+    FactoryGirl.create(:shipping_method).tap do |sm|
+      sm.zones = [@zone]
```
Comment (jordan-brough) [inline core/lib/spree/testing_support/order_walkthrough.rb]: might as well pass this into `create` to begin with, right?

**Comment by jordan-brough** [comment]
:+1:

File: core/lib/spree/testing_support/order_walkthrough.rb (hunk 1 above)
Comment (jhawthorn) [inline core/lib/spree/testing_support/order_walkthrough.rb]: Good catch. Made this change.

**Pull request merged**